packages/amplify-provider-awscloudformation/src/utils/upload-auth-trigger-template.ts

```diff
@@ -0,0 +1,40 @@
+import { $TSContext, $TSObject, FeatureFlags, JSONUtilities, pathManager } from 'amplify-cli-core';
+import { S3 } from '../aws-utils/aws-s3';
+import path from 'path';
+import fs from 'fs-extra';
+import * as constants from '../constants';
+
+export const AUTH_TRIGGER_TEMPLATE = 'auth-trigger-cloudformation-template.json';
+export const AUTH_TRIGGER_STACK = 'AuthTriggerCustomLambdaStack';
+export async function uploadAuthTriggerTemplate(context: $TSContext) {
+  if (!FeatureFlags.getBoolean('auth.breakCircularDependency')) {
+    return {};
+  }
+  const category = 'auth';
+  let { amplifyMeta } = context.amplify.getProjectDetails();
+  const authResource = amplifyMeta?.auth ?? {};
+  const authTriggerCfnFilePath = path.join(pathManager.getBackendDirPath(), category, Object.keys(authResource)[0], AUTH_TRIGGER_TEMPLATE);
+  if (fs.existsSync(authTriggerCfnFilePath)) {
```

Comment:
Instead of checking for the file existence here, you could just try to read it, and if an error occurs, return `{}`.